alphagov/notifications-api
# Stop letters when service is in trial

## What - 

If a service is in trial mode we don't want to send letters out as the cost per letter is significant.
This is a small story - https://www.pivotaltracker.com/story/show/150487327

- No letters are sent on v2 endpoint in trial mode
- v1 endpoint also updated to not send letters although this should be prevented by admin
- letter jobs for services in trial are not processed but given error status
- if the service is in research mode then put the job on a research queue to simulate letter sent

## How to review - 

- Set service to trial mode, you should not be able to send a letter notification

&nbsp;&nbsp; set up by uploading a letter job when live, then switch the service to trial 
&nbsp;&nbsp;&nbsp;&nbsp;- send a letter manually via letter jobs
&nbsp;&nbsp;&nbsp;&nbsp;- to test the letter job schedule update the config.py 

```
        'run-letter-jobs': {
            'task': 'run-letter-jobs',
            'schedule': crontab(minute='*/1'),
...
```
&nbsp;&nbsp;&nbsp;&nbsp; to run it every minute rather than just at 5:30pm
- Letter jobs sent in trial mode will report an error message
- Set service to research mode, no dvla files are created in S3, letter job status instantly set to `send to dvla`
